lms/djangoapps/email_marketing/tasks.py

```diff
@@ -120,7 +123,13 @@ def _create_sailthru_user_parm(sailthru_vars, email, new_user, email_config):
 
     # if new user add to list
     if new_user and email_config.sailthru_new_user_list:
-        sailthru_user['lists'] = {email_config.sailthru_new_user_list: 1}
+        user_list = json.loads(email_config.sailthru_new_user_list)
+        user_list_name = user_list.get(site_domain)
+        if user_list_name:
+            sailthru_user['lists'] = {user_list_name: 1}
+        else:
+            site = Site.objects.get_current()
+            sailthru_user['lists'] = {user_list.get(site.domain): 1}
```

Comment:
The statement `user_list.get(site.domain)` can return null. I don't think we should have a `null` value as the key here. Let's check first whether `user_list.get(site.domain)` return is not null first.